### ISSUE TITLE
Feature request: common queue for different project

### ISSUE THREAD

**Issue by guldmann** [MargaretKrutikova/DeployServiceWebApi#2]
In cases where two or more project will conflict if run simultaneous, it would be nice to have the jobbs added to a queue "FIFO" .

Second best solution would be the ability to mark projects as conflicting, and then disable conflicting project until running process is done.

**Comment by MargaretKrutikova** [comment]
Yes, I was thinking about implementing some kind of a queue. On the other hand, there are projects that won't conflict, so they don't have to wait in the same queue, so I guess there should be many queues for projects conflicting with each other, that way it will be more efficient. 
Have you encountered problems when running the deploy service with conflicting projects?

**Comment by MargaretKrutikova** [comment]
Now when I think about it, isn't it like all services that belong to a project are conflicting? So they can have their own queue, I guess?

**Comment by anderejd** [comment]
I would like the ability to simply tag each script/service/thing (let's call them Thing for now) with an optional queue name/id and the appropriate number of queues would be created for me. If a Thing does not have the queue specified it will be allowed to run in parallel with all other Things.
Would that work?

**Comment by guldmann** [comment]
No real conflicting project yet.
I like @anderejd  suggestion.

**Comment by MargaretKrutikova** [comment]
what if you have really a lot of scripts in the settings file? you will then have to remember which queue id you assigned to one script, remember which scripts are conflicting and assign the same id to them. isn't it easier to utilize the concept of "project" to define conflicting scripts. What would be the advantage of tagging scripts with a queue id over using projects for the same purpose?

**Comment by guldmann** [comment]
Makes sense to have "project" as the grouping of conflicting scripts. I can see how adding queue on script level could be error prone.

**Comment by anderejd** [comment]
Grouping is fine by me too. (instead of tagging)
I think "Project" is too vague though. In our use-case most Things™ in the same config file will be part of the same project (business perspective).
How about calling the queue grouping "Queue"? :) EDIT: Or something else with the word "queue" in it?

**Comment by MargaretKrutikova** [comment]
Yes, I agree. "Project" is a bit misleading, I like changing project-thing into queue-thing. I will start with the implementation :) Thanks for the suggestions!

**Comment by MargaretKrutikova** [comment]
Long time no see ;) I have implemented basic queues for services that belong to the same "group" which is converted into a queue internally. Project™ is renamed into Group™. I updated the client project according to the api changes. Let me know if there is anything else you would like to have fixed/implemented :)

**Comment by anderejd** [comment]
Sweet, I will update and try it out later this week!
